src/components/vm/snapshots/vmSnapshotsCreateModal.jsx

```diff
@@ -87,29 +128,42 @@
     }
 
     onValidate(submitted = false) {
-        const { name } = this.state;
-        const { vm } = this.props;
+        const { name, memoryPath } = this.state;
+        const { vm, isExternal } = this.props;
         const validationError = {};
 
         if (vm.snapshots.findIndex(snap => snap.name === name) > -1)
             validationError.name = _("Name already exists");
         else if (!name && submitted)
             validationError.name = _("Name should not be empty");
 
+        if (isExternal && vm.state === "running" && !memoryPath)
+            validationError.memory = _("Memory file can not be empty");
+
         return validationError;
     }
 
     onCreate() {
         const Dialogs = this.context;
-        const { vm } = this.props;
-        const { name, description } = this.state;
+        const { vm, isExternal, storagePools } = this.props;
+        const { name, description, memoryPath } = this.state;
+        const disks = Object.values(vm.disks);
         const validationError = this.onValidate(true);
 
         this.setState({ submitted: true });
 
         if (!Object.keys(validationError).length) {
             this.setState({ inProgress: true });
-            snapshotCreate({ connectionName: vm.connectionName, vmId: vm.id, name, description })
+            snapshotCreate({
+                connectionName: vm.connectionName,
+                vmId: vm.id,
+                name,
+                description,
+                memoryPath: vm.state === "running" && memoryPath,
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1280-20240129-095224-b2f949bc-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)